src/plugins/data_source_management/public/components/utils.ts

```diff
@@ -303,6 +304,23 @@ export const handleDataSourceFetchError = (
   });
 };
 
+export const handleDataSourceViewError = (
+  changeState: (state: { showError: boolean }) => void,
+  notifications: ToastsStart,
+  failedDataSourceId: string,
+  defaultDataSource: string | null,
+  callback?: () => Promise<void>
+) => {
+  changeState({ showError: true });
+  notifications.add({
+    title: i18n.translate('dataSource.dataSourceUnavailableError', {
+      defaultMessage: `Data source ${failedDataSourceId} is not available `,
+    }),
+    text: defaultDataSource && callback ? toMountPoint(switchDefaultButton(callback)) : '',
```

Comment:
is it possible that callback is undefined or null?

Reply:
Possible to have the callback undefined or null when not pass it. The callback here is set to optional referring to `handleDataSourceFetchError`, defaultDataSource can be null since we might not have it set up from uiSettings. The case here is: when default data source exist, the callback has to exist, when no default data source, there is no need for the callback. 
I agree to set it mandatory and only use defaultDataSource as condition check

Reply:
synced offline, the defaultDataSource should be the only condition check